### PR TITLE
Gallery Update

### PR DESCRIPTION
**Pull request by toropeza**
This is a pull request for changes to the AppInventor Gallery. A working version can be found at the following URL.
https://1-dot-aigallerythomas.appspot.com/#5769015641243648

Things Changed:
-A better layout for profile pages in the gallery has been added to where the user can see a preview of apps then click to see more
-Fixed a bug where double clicking on an app in the gallery would open duplicate apps on the right of the app view page
-Added the tutorial tab and the ability to make apps tutorial apps in the same way apps are featured

The changes were organized by Thomas Oropeza (thomasoropeza@gmail.com) and David Wolber.

### CODE REVIEW

File: appinventor/appengine/war/WEB-INF/appengine-web.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (jisqyv) [inline appinventor/appengine/war/WEB-INF/appengine-web.xml]: Back out these changes. DO NOT EDIT THIS FILE to change bucket configurations!

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java [hunk 2]
```diff
@@ -966,10 +966,10 @@ private YaVersion() {
   // key as the Companion it is replacing, as the Package Manager
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: DO NOT EDIT THIS FILE. Back out these changes.

File: appinventor/components/src/com/google/appinventor/components/runtime/Form.java [hunk 3]
```diff
@@ -342,18 +342,19 @@ public void onGlobalLayout() {
       Log.d(LOG_TAG, "keyboard hidden!");
       if (keyboardShown) {
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/runtime/Form.java]: Presumably you didn't make these changes. Merge master in.

File: appinventor/aiplayapp/youngandroidproject/project.properties [hunk 4]
```diff
@@ -7,12 +7,12 @@ icon=AI2Companion-Icon.png
 # The Version code is an integer. Each new version of an App uploaded
```
Comment (jisqyv) [inline appinventor/aiplayapp/youngandroidproject/project.properties]: This file shouldn't change in your PR. Make sure you merge in master.

File: appinventor/appengine/src/com/google/appinventor/client/GalleryClient.java [hunk 5]
```diff
@@ -157,7 +158,30 @@ public void onSuccess(GalleryAppListResult appsResult) {
     // This is below the call back, but of course it is done first
     ode.getGalleryService().getFeaturedApp(start, count, callback);
   }
- /**
+/**
+  * GetTutorial gets tutorial apps, currently unimplemented
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/GalleryClient.java]: Is this comment still accurate.

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryToolbar.java [hunk 6]
```diff
@@ -57,7 +58,7 @@ public void onClick(ClickEvent event) {
         GalleryClient.getInstance().FindApps(searchText.getText(), 0, GalleryList.NUMAPPSTOSHOW, 0, true);
         searchText.setFocus(true);
         Ode.getInstance().switchToGalleryView();
-        GalleryListBox.getGalleryListBox().getGalleryList().setSelectTabIndex(3); //TODO change this to constant
+        GalleryListBox.getGalleryListBox().getGalleryList().setSelectTabIndex(SEARCHTAB); //TODO change this to constant
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryToolbar.java]: Update comment

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryToolbar.java [hunk 7]
```diff
@@ -73,7 +74,7 @@ public void onKeyDown(KeyDownEvent e) {
           GalleryClient.getInstance().FindApps(searchText.getText(), 0, GalleryList.NUMAPPSTOSHOW, 0, true);
           searchText.setFocus(true);
           Ode.getInstance().switchToGalleryView();
-          GalleryListBox.getGalleryListBox().getGalleryList().setSelectTabIndex(3); //TODO change this to constant
+          GalleryListBox.getGalleryListBox().getGalleryList().setSelectTabIndex(SEARCHTAB); //TODO change this to constant
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/GalleryToolbar.java]: Update comment

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java [hunk 8]
```diff
@@ -524,4 +540,190 @@ private boolean validEmailFrequency(String emailFrequencyString){
 
     return true;
   }
+
+  /**
```
Comment (jisqyv) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java]: Didn't this class already exist? Why is it added here?

**Comment by jisqyv** [comment]
@toropeza You need to address the line comments above. Also merge in master and deal with the conflicts.

**Comment by toropeza** [comment]
@wolberd @jisqyv Thanks, i'll let you know when i'm finished.

File: appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java (hunk 8 above)
Comment (toropeza) [inline appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/ProfilePage.java]: @jisqyv @wolberd I believe that whenever classes need to implement this class it is added as a private wrapper class. In this case it is added to hold the content of the catalog

**Comment by toropeza** [comment]
Added new commits. Working version can now be found at 
https://1-dot-aigallerythomas.appspot.com/#5769015641243648
@jisqyv @wolberd

**Comment by jisqyv** [comment]
@toropeza @wolberd LGTM, On Gerrit.

**Comment by jisqyv** [comment]
MERGED

**Pull request closed without merging**